unity-renderer/Assets/Scripts/MainScripts/DCL/Components/LoadableShapes/NFTShape/NFTShapeLoaderController.cs

```diff
@@ -150,6 +150,10 @@ private void FetchNFTContents()
 
     private void FetchNFTInfoSuccess(NFTInfo nftInfo)
     {
+        // the FetchNFTInfoSuccess action can get called at the NFTInfoLoadHelper after destroying this controller
```

Comment:
Not happy with this fix, but approving for stability. The best way to fix this shouldn't be to ensure the subscriptor is unsubscribed from this event when is destroyed?

Reply:
I thought about that as well but the thing with this `FetchNFTInfoSuccess()` "Action" is that it's not used through event subscription, it's being injected into another class method as a callback to be invoked when successfull: https://github.com/decentraland/unity-renderer/blob/074d7043c2eacdc262eddf9252236d374620031e/unity-renderer/Assets/Scripts/MainScripts/DCL/Components/LoadableShapes/NFTShape/NFTShapeLoaderController.cs#L148

And even when this `NFTShapeLoaderController` becomes null when being destroyed, that callback injected of course can still be invoked afterwards

Reply:
Let me see if I can make a little refactor to solve that, since I think each `NFTShapeLoaderController` uses its own `NFTInfoLoadHelper` we should be able to add an event in the helper and unsubscribe the controller when destroyed

Reply:
Done!